### DISCUSSION TITLE
Why do the dependencies need to be reinstalled everytime i set my policy to `build.sanitizer.address`?

### DISCUSSION THREAD

**Discussion by mirageAlchemy** [xmake-io/xmake#5805] [Q&A] [answered]
This is not neccesarily a bug so I did not post an issue. For me everytime I switch the mode and `set_policy("build.sanitizer.address", true)`, all my dependencies are reinstalled, even if some of them are header only. I am not entirely against it because you indeed need to rebuild it for some source code included lib, but downloading it all over again seems cumbersome.

**Comment by star-hengxing** [comment, marked as answer]
Asan requires that all code (including third party libraries) is built with asan flags. Otherwise, false positives will occur.

xrepo doesn't download again, it will cache the source code archive.

**Reply by mirageAlchemy** [reply]
Thanks for the apply, I was just doubtful about the header only library part, but I guess xmake does not make a difference between header only lib and lib that need to be linked. It is good to know the source code is cached.
